backend/server.js

```diff
@@ -1,29 +1,177 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-import mongoose from 'mongoose'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/authAPI"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
+import express from "express";
+import bodyParser from "body-parser";
+import cors from "cors";
+import mongoose from "mongoose";
+import crypto from "crypto";
+import bcrypt from "bcrypt-nodejs";
+
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/authAPI";
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
+
+const userSchema = new mongoose.Schema({
+  name: {
+    type: String,
+    required: true,
+  },
+  email: {
+    type: String,
+    unique: true,
+    required: true,
+    minlength: 5,
+  },
+  password: {
+    type: String,
+    required: true,
+    minlength: 8,
+  },
+  accessToken: {
+    type: String,
+    unique: true,
+  },
+});
+
+userSchema.pre("save", async function (next) {
+  const user = this;
+
+  if (!user.isModified("password")) {
+    return next();
+  }
+
+  const salt = bcrypt.genSaltSync();
+  user.password = bcrypt.hashSync(user.password, salt);
+  next();
+});
+
+const User = mongoose.model("User", userSchema);
+
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+const port = process.env.PORT || 8080;
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(bodyParser.json());
+
+const authenticateUser = async (req, res, next) => {
+  try {
+    const accessToken = req.header("Authorization");
+    const user = await User.findOne({ accessToken });
+    if (!user) {
+      throw "User not found";
+    }
+    req.user = user;
+    next();
+  } catch (err) {
+    const errorMessage = "Please try logging in again";
+    console.log(errorMessage);
+    res.status(401).json({ error: errorMessage });
+  }
+};
+
+// Sign-up
+app.post("/users", async (req, res) => {
+  try {
+    const { name, email, password } = req.body;
+    const user = await new User({
+      name,
+      email,
+      password,
+    });
```

Comment:
You set the `accessToken` field as required, without the default value and didn't provide the token here... this is why your app isn't working.

Reply:
I don't want the user to be auto loged in when signing up that is why I don't provide the accesstoken here